layers: update device tree blob list

shawaj opened the pull request:
Add new device tree blobs to the list to bring in support for new hardware such as merus-amp and more. From https://github.com/raspberrypi/linux/tree/rpi-5.4.y/arch/arm/boot/dts/overlays

Change-type: minor
Signed-off-by: Aaron Shaw <aaron@balena.io>

resin-jenkins commented:
Can one of the admins verify this patch?

shawaj commented:
Based on [this overlay_map list](https://github.com/raspberrypi/linux/blob/rpi-5.4.y/arch/arm/boot/dts/overlays/overlay_map.dts) there are some deprecated device tree blobs and a number which are platform specific, which are not reflected in our layer.conf

Shall I update this PR to reflect the information there?

acostach commented:
@resin-jenkins test this please

shawaj commented:
@acostach think I missed a backslash on line 202. Will amend.

What did you think about the overlay_map thing?

shawaj commented:
@resin-jenkins test this please

acostach commented:
@resin-jenkins test this please

shawaj commented:
@acostach fixed the missing slash now. Let me know what you think about dealing with the overlay_map list in respect to the different machines?

acostach commented:
@resin-jenkins test this please

acostach commented:
thanks @shawaj , I don't know much about what the overlay_map does, I'll have to look into it to give an opinion.

shawaj commented:
@acostach I believe the overlay map list basically shows some overlays that are deprecated or only valid on certain devices.

For example, the below shows i2c6 only for bcm2711, a deprecated one and a renamed one:
```
i2c6 {
		bcm2711;
	};

	lirc-rpi {
		deprecated = "use gpio-ir";
	};

	pi3-act-led {
		renamed = "act-led";
	};
```

My thought was to remove any deprecated and renamed ones and for the bcm2711 specific ones only have those on the pi 4 builds.

What do you think?

floion commented:
@resin-jenkins test this please

floion commented:
@shawaj at this moment only the rpi4 is on kernel 5.4
Is your intention to add these overlays for rpi4 or for which boards?

shawaj commented:
@floion the intention is to add these overlays for all Pi versions

acostach commented:
@resin-jenkins test this please

shawaj commented:
@acostach does the Jenkins bot give any output as to why a build is failing? I clicked on details but it requires a login :-)

acostach commented:
@shawaj the Pi4 and 400 are on the 5.4 kernel and the others are on the previous version 4.19 where some of the dtbos you add don't exist. But the plan is that we update all boards to 5.4.y

floion commented:
Superseded by https://github.com/balena-os/balena-raspberrypi/pull/587
